### ISSUE TITLE
Passing `DedicatedQDQPair=True` raises `Error: two nodes with same node name`

### ISSUE THREAD

**Issue by fxmarty** [microsoft/onnxruntime#14142]
### Describe the issue

I need to pass `DedicatedQDQPair=True` to quantize a model for TensorRT, as recommended: https://github.com/microsoft/onnxruntime/blob/4eac0db3afecf93063bf3d4ff289763453aaf89f/onnxruntime/python/tools/quantization/qdq_quantizer.py#L100-L104

However, the error `Error: two nodes with same node name` is raised once I try to load the model in InferenceSession post quantization.

When I pass `DedicatedQDQPair=False`, I don't have the error.

### To reproduce

The model can be found at (uploading): https://huggingface.co/fxmarty/bugged-quantized-model-DedicatedQDQPair/tree/main

The quantization config used is: https://huggingface.co/fxmarty/bugged-quantized-model-DedicatedQDQPair/blob/main/ort_config.json

Run:

```python
import onnxruntime

session_options = onnxruntime.SessionOptions()
session_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_DISABLE_ALL

session = onnxruntime.InferenceSession("model_quantized.onnx", providers=["CPUExecutionProvider"], sess_options=session_options)
```

This will raise:
```
Fail: [ONNXRuntimeError] : 1 : FAIL : Load model from /tmp/quantized_distilbert_sst2_qdq_everywhere_symmetric/model_quantized.onnx failed:This is an invalid 
model. Error: two nodes with same node name (/distilbert/embeddings/LayerNorm/Add_1_output_0_QuantizeLinear).
```

Example of duplicate node names (the two have the same name):

![image](https://user-images.githubusercontent.com/9808326/210774114-0beedbe9-0a66-453d-945a-659cc049ff3f.png)

Is this expected? I recall it was working at some point.

### Urgency

somewhat urgent

### Platform

Linux

### OS Version

Ubuntu 22.04 jammy, x86_64 Linux 5.15.0-53-generic

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.13.1

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

Default CPU

**Comment by fxmarty** [comment]
In an older model I did not have the issue (nodes named `onnx::Shape_244_QuantizeLinear_2` and `onnx::Shape_244_QuantizeLinear_1`, while the output name from the `Add` node is `onnx::Shape_244`):

![image](https://user-images.githubusercontent.com/9808326/210776959-b8fb0f87-3361-4fdc-a12e-5ba840798717.png)

**Comment by stevenlix** [comment]
It seems a bug. Thanks for reporting it. By the way, there should be no problem to share one pair of QDQ nodes from the same node output. Do you see any issue by not setting DedicatedQDQPair?

**Comment by fxmarty** [comment]
Thank you for your quick reply! I have no issue when DedicatedQDQPair=False.

**Comment by fxmarty** [comment]
@stevenlix Let me know if I can be of any help to solve this issue, happy to submit a PR.

**Comment by cyang49** [comment]
Any updates on this issue?

**Comment by HYLcool** [comment]
Hi, it's been fixed in PR #14258
You can try it using the latest code.

**Comment by fxmarty** [comment]
Awesome, thanks for the fix! Hope it will be in 1.14.
